lib/string_decoder.js

```diff
@@ -185,10 +199,10 @@ function utf8FillLast(buf) {
   if (r !== undefined)
     return r;
   if (this.lastNeed <= buf.length) {
-    buf.copy(this.lastChar, p, 0, this.lastNeed);
-    return this.lastChar.toString(this.encoding, 0, this.lastTotal);
+    copy(buf, this.lastChar, p, 0, this.lastNeed);
+    return this.lastChar.utf8Slice(0, this.lastTotal);
```

Comment:
I think this can be `utf8Slice(this.lastChar, 0, this.lastTotal)` to fit with the other changes?